### DISCUSSION TITLE
How to trigger different action only on merge

### DISCUSSION THREAD

**Discussion by danielinclouds** [community/community#25354] [Actions]
<p>Hey,</p>
<p>I’m having a problem implementing my workflow in a single action file:</p>
<p>I want to run:</p>
<p>* only step1 on every push to feature branch (other than master)</p>
<p>* only step1 on a pull request</p>
<p>* step1 and step2 on merge</p>
<p>This is my pipeline:</p>
<pre><code>name: Test

on:
  push:
    branches-ignore:
      - master
  pull_request:

jobs:

  build:
    name: Testing
    runs-on: ubuntu-latest
    steps:

    - name: Step 1
      run: |
        echo " Step 1"

    - name: Step 2
      if: github.event.pull_request.merged == true
      run: |
        echo " Step 2"
</code></pre>
<p>I can implement my requirements with 2 files but I would like to keep it in one if that is possible.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/danielinclouds">@danielinclouds</a> ,</p>
<p>Merge pull request will not trigger the workflow. By default, the pull_request’s activity type is opened, synchronize, and reopened. Hence you need to add ‘closed’ type for pull request event. Code sample as below:</p>
<pre><code>name: Test

on:
  push:
    branches-ignore:
      - master
&lt;font color="#FF6600"&gt; pull_request:
    types:
      - opened
      - reopened
      - synchronize
      - closed&lt;/font&gt;jobs:

  build:
    name: Testing
    runs-on: ubuntu-latest
    steps:
      - name: Dump GitHub context
        env:
          GITHUB_CONTEXT: ${{ toJson(github) }}
        run: echo "$GITHUB_CONTEXT"

      - name: Step 1
        run: |
          echo " Step 1"

      - name: Step 2
        if: github.event.pull_request.merged == true
        run: |
          echo " Step 2"
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181090580-c0b904b9-0866-47f3-b150-6f0754bf5365.png" alt="mergepr.png" data-base62-sha1="bOvZHbzK1mKhVc8VjSOceO7x2rA" width="542" height="293"></p>
<p>Hope it’s clear and helpful!</p>

**Comment by danielinclouds** [comment]
<p>Thanks, <a class="mention" href="/weide-zhou">@weide-zhou</a> that was exactly what I was looking for.</p>

**Comment by repo-matic** [comment]
<p>Thanks! I’d been looking for this for a while.</p>

**Comment by Vadorequest** [comment]
<aside class="quote group-github-partners" data-username="weide-zhou" data-post="2" data-topic="18260">
<div class="title">
<div class="quote-controls"></div>
 weide-zhou:</div>
<blockquote>
<p><code>&lt;font color="#FF6600"&gt; </code></p>
</blockquote>
</aside>
<p>There are some styling that mess up with the readability of your answer :slight_smile:</p>

**Comment by weide-zhou** [comment]
<p>Thanks <a class="mention" href="/Vadorequest">@Vadorequest</a>, this is caused by the community forum update, i tried but cannot edit the code. You can ignore the styling code like <code>&lt;font&gt;&lt;/font&gt;</code> :joy:.</p>
